src/coreclr/jit/unwindarm.cpp

```diff
@@ -649,8 +679,19 @@ void Compiler::unwindEmitFunc(FuncInfoDsc* func, void* pHotCode, void* pColdCode
     }
 #endif // FEATURE_CFI_SUPPORT
 
-    func->uwi.Allocate((CorJitFuncKind)func->funKind, pHotCode, pColdCode, true);
+    if (func->funKind == FUNC_ROOT)
+    {
+        func->uwi.Allocate((CorJitFuncKind)func->funKind, pHotCode, pColdCode, true);
+    }
+    else
+    {
+        // If the function is split, EH funclets are always cold.
+        const bool isHotCode = (fgFirstColdBlock == nullptr);
+        func->uwi.Allocate((CorJitFuncKind)func->funKind, pHotCode, pColdCode, isHotCode);
+    }
 
+    // func->uwiCold is used only for the cold part of the main function,
+    // and is always NULL for funclets.
```

Comment:
There's something confusing about the model here. In particular, it looks like cold funclets have their unwind info in the "hot" `uwi` structure instead of having an empty `uwi` and putting their unwind info in the `uwiCold` structure.

Wouldn't it be better to introduce a notion of the "hot" unwind info (in `uwi`) being empty, if the entire funclet is cold?

One of the things I'm worried about here is distributing the logic about "if a function with EH is split then all funclets are cold" instead of just setting up the FuncInfoDsc with the right information and simply iterating over it.

Reply:
Good point. I was thinking it would make more sense to use `uwiCold` for cold EH funclets, but the current implementation assumes `uwi != nullptr` is never true, so this may require some non-trivial refactoring. If we're okay with the higher churn, I can try this out.